platform-sdk/swirlds-config-api/src/main/java/com/swirlds/config/api/observable/ObservableConfiguration.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.swirlds.config.api.observable;
+
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Consumer;
+
+/**
+ * Extension of the {@link Configuration} API to add support for observable properties ({@link ObservableProperty}) that
+ * might change at runtime.
+ * <p>
+ * This is only created for discussing the API. Later the API might become a part of {@link Configuration}.
+ */
+public interface ObservableConfiguration extends Configuration {
+
+    /**
+     * Returns an observable property based on the given {@code key}. Calling this method will never return {@code null}
+     * or throw an {@link java.util.NoSuchElementException}. Even if the property is currently not defined by the config
+     * it might be present in future. Therefore an {@link ObservableProperty} instance is always returned that can be
+     * used to check the state and value of a property at runtime.
+     *
+     * @param key  the key of the property
+     * @param type the value type of the property
+     * @param <T>  the value type of the property
+     * @return the observable property
+     */
+    @NonNull
+    <T> ObservableProperty<T> getObservableProperty(@NonNull String key, @NonNull Class<T> type);
```

Comment:
I don't like the idea of referencing the property my a key, because we moved away from that model. Can we make is so that it is a variable inside a config, like so:
```
public record SomeConfig(
  @ConfigProperty(defaultValue = "a value")
  ObservableProperty<String> someProperty
)
```
This is more in line with our current approach

Reply:
my bad, this is the way it already works